operator/pkg/watch/restricted_enqueue_request_for_owner.go

```diff
@@ -0,0 +1,227 @@
+package watch
```

Comment:
Maybe I missed something, but I still so not see a good enough reason to implement the EventHandler interface. We can instead use the inbuild handler.Funcs{
UpdateFunc: ..
CreateFunc: ..
DeleteFunc: ..
GenericFunc: .. (not required)
}